server/cpho/views/user_management.py

```diff
@@ -47,3 +65,197 @@ def get_context_data(self, **kwargs):
+            attrs={
+                "id": "phac_org_multi__textinput",
+            },
+        ),
+        label=tdt("phac_org_multi"),
+    )
+
+
+class CreateUserForm(UserForm):
+    email = forms.EmailField(
+        required=True,
+        widget=forms.EmailInput(
+            attrs={
+                # Only allow emails ending in @*.gc.ca or @canada.ca
+                "pattern": r"^[a-zA-Z0-9_.+\-]+@([a-zA-Z0-9\-]+\.gc\.ca|canada\.ca)$",
+                "oninvalid": f"setCustomValidity('{tdt('invalid email use phac-aspc.gc.ca or canada.ca')}')",
+                "oninput": "setCustomValidity('')",
+                "class": "form-control",
+            }
+        ),
+    )
+
+    def clean_email(self):
+        # force only lowercase emails, safer to compare that way
+        email = self.cleaned_data["email"].lower()
+        if not (email.endswith("@canada.ca") or email.endswith(".gc.ca")):
+            raise ValidationError(tdt("email_exception"))
+
+        if User.objects.filter(email__icontains=email).exists():
+            raise ValidationError(tdt("account_for_email_already_exists"))
+
+        return email
+
+
+class UserFormView(FormView, CanManageUsersMixin):
+    template_name = "user_management/create_modify_user.jinja2"
+
+    def form_valid(self, form):
+        self.save_form(form)
+        self.show_message()
+        return super().form_valid(form)
+
+    def show_message(self):
+        raise NotImplementedError()
+
+    def get_success_url(self):
+        return reverse("manage_users")
+
+    def get_or_create_user(self, form):
+        raise NotImplementedError()
+
+    def save_form(self, form):
+        with transaction.atomic():
+            user = self.get_or_create_user(form)
+            self.assign_orgs(user, form)
+            self.assign_groups(user, form)
+
+    def assign_orgs(self, user, form):
+        user.phac_org_roles.all().delete()
+        for org in form.cleaned_data["phac_org_multi"]:
+            PhacOrgRole.objects.create(user=user, phac_org=org)
+
+    def assign_groups(self, user, form):
+        user.groups.remove(
+            GroupFetcher.account_manager_group,
+            GroupFetcher.hso_group,
+            GroupFetcher.admin_group,
+        )
+
+        if form.cleaned_data["is_admin"]:
+            user.groups.add(GroupFetcher.admin_group)
+
+        if form.cleaned_data["is_account_manager"]:
+            user.groups.add(GroupFetcher.account_manager_group)
+
+        if form.cleaned_data["is_hso"]:
+            user.groups.add(GroupFetcher.hso_group)
+
+
+class CreateUser(UserFormView, CanManageUsersMixin):
+    def get_form(self):
+        if self.request.POST:
+            return CreateUserForm(self.request.POST)
+        else:
+            return CreateUserForm()
+
+    def get_or_create_user(self, form):
+        email = form.cleaned_data["email"]
+        return User.objects.create_user(
+            email=email, username=email, password=email
```

Comment:
we can stick with password=username=email for user testing

Reply:
I agree that for now we can keep username=passowrd
but just wanted to discuss if username= email is the best idea..
as of iright now we have other users like cdsb_lead, cdsb_user etc from the seed script, whose username does not end with @canada.ca etc.
Could this create confusion?